packages/cli/test/integration/publicApi/credentials.test.ts

```diff
@@ -0,0 +1,313 @@
+import express from 'express';
+import { UserSettings } from 'n8n-core';
+import { Db } from '../../../src';
+import { randomName, randomString } from '../shared/random';
+import * as utils from '../shared/utils';
+import type { CredentialPayload, SaveCredentialFunction } from '../shared/types';
+import type { Role } from '../../../src/databases/entities/Role';
+import type { User } from '../../../src/databases/entities/User';
+import * as testDb from '../shared/testDb';
+import { RESPONSE_ERROR_MESSAGES } from '../../../src/constants';
+
+jest.mock('../../../src/telemetry');
+
+let app: express.Application;
+let testDbName = '';
+let globalOwnerRole: Role;
+let globalMemberRole: Role;
+let saveCredential: SaveCredentialFunction;
+
+beforeAll(async () => {
+	app = await utils.initTestServer({ endpointGroups: ['publicApi'], applyAuth: false });
+	const initResult = await testDb.init();
+	testDbName = initResult.testDbName;
+
+	utils.initConfigFile();
+
+	globalOwnerRole = await testDb.getGlobalOwnerRole();
+	globalMemberRole = await testDb.getGlobalMemberRole();
+	const credentialOwnerRole = await testDb.getCredentialOwnerRole();
+	saveCredential = affixRoleToSaveCredential(credentialOwnerRole);
+
+	utils.initTestLogger();
+	utils.initTestTelemetry();
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['User', 'SharedCredentials', 'Credentials'], testDbName);
+});
+
+afterAll(async () => {
+	await testDb.terminate(testDbName);
+});
+
+test('POST /credentials should create credentials', async () => {
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
```

Comment:
Utils.createUser() should replace both lines. It creates a member and automatically assigns an API key

Reply:
yes it does, but I think it shouldn't

also we test here the case when we don't have multiple users, but just an uninitialized owner

Reply:
Makes sense